caabernathy/YourWord
# Add functionality to search keywords to get scripture

Related to #17 

Felt like picking scriptures by book, chapter, verses wasn't enough. Need a way for a user who doesn't know this info to search for scripture verse by keyword. This adds that functionality.



Now users can go to the "Your Verses" screen where the default is search by verse. To simplify the UI, the select by verses is only shown if the user selects that option. Searching by keyword shows a list of the verses and the user can tap on one to add it to their list.

To support this feature, the API has added an endpoint from Bolls Life API to provide the backend functionality.



**Test plan (required)**
1. Open app and tap on Your Verses
2. Enter a text in the text field. ex: "Test"
3. View results and verify that the keyword is in each result
4. Tap one of the results
5. Verify that you now have the scripture available to memorize
6. Tap on the Pick by Verses option and verify that this flow works as before
7. Test error flows. ex: locally stopping the server and verifying a toast error is shown.

Demonstrate the code is solid. Example: Screenshots / videos are helpful if the pull request changes UI.
<img width="447" alt="Screenshot 2024-07-14 at 2 06 33 AM" src="https://github.com/user-attachments/assets/b86597b3-c043-4905-8500-8ec62f75a9b0">
![Simulator Screenshot - iPhone 15 Pro - 2024-07-14 at 02 06 52](https://github.com/user-attachments/assets/6c9c99a5-0a32-49dc-9cb4-be535dfbd234)
![Simulator Screenshot - iPhone 15 Pro - 2024-07-14 at 02 07 01](https://github.com/user-attachments/assets/ad149ec8-8734-46ec-9827-613468fd8a77)
